### PR TITLE
bmx: Add version 1.3

### PR DESCRIPTION
**Pull request by irieger**
### Summary
Changes to recipe:  **bmx/1.3**

#### Motivation
There is a ton of improvements - new features as well as a number of bug fixes - since the last official 1.2 release.

#### Details
https://github.com/bbc/bmx/compare/v1.2...v1.3

A ton of features and fixes including some build system improvements we suggested have been added.
Also note, the `1.3-cmake-fixes.patch` matches `cci.20240517-cmake-fixes.patch`, I just made a copy in this case as I understand the cci.* version more like a placeholder and felt it would be best to align the patch to the actual version.

Also a question I asked in Slack too for the reviewers: Best keep the *cci.20240517* version or remove it? I'd tend to remove it but wanted to ask first.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by AbrilRBS** [comment]
> Also a question I asked in Slack too for the reviewers: Best keep the cci.20240517 version or remove it? I'd tend to remove it but wanted to ask first.

If there's no usage in CCI, we can remove it. Note that the binaries will still be available in Conan, it just won't get any updates even when the recipe changes

**Review by AbrilRBS — DISMISSED** [review]
Looks great, thanks a lot!

**Comment by irieger** [comment]
> > Also a question I asked in Slack too for the reviewers: Best keep the cci.20240517 version or remove it? I'd tend to remove it but wanted to ask first.
> 
> If there's no usage in CCI, we can remove it. Note that the binaries will still be available in Conan, it just won't get any updates even when the recipe changes

Ok, perfect. I knew that the binaries and recipe still are available when specified by version, was just unsure whether we remove it directly.

I pushed another changes as the patch for the CCI version is no longer needed now.

**Comment by AbrilRBS** [comment]
Grat catch, thanks! Feel free to ping me once the release is published :)

File: recipes/bmx/all/conandata.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
 sources:
+  "1.3":
+    # Todo replace with actual 1.3 release in the next days
+    url: "https://github.com/bbc/bmx/archive/refs/tags/v1.3.tar.gz"
```
[the remark this answers is not in the document]
Comment (irieger) [reply recipes/bmx/all/conandata.yml]: Changed it. For 1.3 at least. The existing 1.2 we keep as is, right?

File: recipes/bmx/all/conandata.yml [hunk 2]
```diff
@@ -1,11 +1,16 @@
 sources:
+  "1.3":
+    # Todo replace with actual 1.3 release in the next days
```
Comment (AbrilRBS) [inline recipes/bmx/all/conandata.yml:3]:
```suggestion
```

**Review by AbrilRBS — APPROVED** [review]

**Review by perseoGI — APPROVED** [review]

**Pull request merged**